python/jupytergis_qgis/jupytergis_qgis/qgis_loader.py

```diff
@@ -477,6 +489,147 @@ def build_uri(parameters: dict[str, str], source_type: str) -> str | None:
+                                int(color[3] * 255),
+                            )
+                        )
+
+                        category_symbol.setOpacity(1.0)
+
+                        category = QgsRendererCategory(
+                            condition[2], category_symbol, str(condition[2])
+                        )
+                        renderer.addCategory(category)
+
+            # Graduated Renderer
+            elif render_type == "Graduated":
+                fill_color_rules = color_params.get("circle-fill-color", [])
+                if (
+                    isinstance(fill_color_rules, list)
+                    and fill_color_rules[0] == "interpolate"
+                ):
+                    ranges = []
+                    for i in range(3, len(fill_color_rules) - 2, 2):
+                        lower_value = fill_color_rules[i]
+                        upper_value = fill_color_rules[i + 2]
+                        color = fill_color_rules[i + 1]
+
+                        if isinstance(color, list) and len(color) == 4:
+                            r, g, b, a = color
+                            qcolor = QColor(int(r), int(g), int(b), int(a * 255))
+                        range_symbol = symbol.clone()
+                        range_symbol.setColor(qcolor)
+                        g_range = QgsRendererRange(
+                            lower_value,
+                            upper_value,
+                            range_symbol,
+                            f"{lower_value} - {upper_value}",
+                        )
+                        ranges.append(g_range)
+                    renderer = QgsGraduatedSymbolRenderer(
+                        symbology_state.get("value"), ranges
+                    )
+
+        elif geometry_type == QgsWkbTypes.LineGeometry:
+            symbol = QgsLineSymbol()
+            color_params = layer_params.get("color", {})
+            color = QColor(color_params.get("stroke-color"))
+            opacity = int(layer_params.get("opacity"))
+            print(f"Line Color: {color.name()}, Opacity: {opacity}")
+            symbol.setColor(color)
+            symbol.setOpacity(opacity)
+            renderer = QgsSingleSymbolRenderer(symbol)
+
+        elif geometry_type == QgsWkbTypes.PolygonGeometry:
+            symbol = QgsFillSymbol()
+            color_params = layer_params.get("color", {})
+            opacity = layer_params.get("opacity", 1.0)
+            symbol.setOpacity(opacity)
+
+            fill_color = QColor(color_params.get("fill-color"))
+            symbol.setColor(fill_color)
+
+            stroke_color = QColor(color_params.get("stroke-color"))
+            stroke_width = color_params.get("stroke-width", 1)
+
+            symbol_layer = symbol.symbolLayer(0)
+            symbol_layer.setStrokeColor(stroke_color)
+            symbol_layer.setStrokeWidth(stroke_width)
+            renderer = QgsSingleSymbolRenderer(symbol)
+
```

Comment:
Maybe we should have a check for `renderer` not being defined here, since if the `render_type` is not supported it will fail here not being able to set the renderer